packages/wdio-firefox-profile-service/src/launcher.js

```diff
@@ -0,0 +1,83 @@
+import Profile from 'firefox-profile'
+import { promisify } from 'util'
+
+export default class FirefoxProfileLauncher {
+    async onPrepare (config, capabilities) {
+        this.config = config
+        this.capabilities = capabilities
+
+        // Return if no profile options were specified
+        if (!this.config.firefoxProfile) {
+            return
+        }
+
+        this.profile = new Profile()
+
+        // Set preferences and proxy
+        this._setPreferences()
+
+        if (!Array.isArray(this.config.firefoxProfile.extensions)) {
+            return this._buildExtension()
+        }
+
+        // Add the extension
+        await promisify(::this.profile.addExtensions)(this.config.firefoxProfile.extensions)
+
+        return this._buildExtension()
+    }
+
+    /**
+     * Sets any preferences and proxy
+     */
+    _setPreferences () {
+        for (const preference in this.config.firefoxProfile) {
```

Comment:
Let's use `Object.entries(this.config.firefoxProfile)` here, not sure how you loop over the object here